django/db/models/base.py

```diff
@@ -454,10 +465,12 @@ def __init__(self, *args, **kwargs):
 
     @classmethod
     def from_db(cls, db, field_names, values):
-        if cls._deferred:
-            new = cls(**dict(zip(field_names, values)))
-        else:
-            new = cls(*values)
+        # TODO - make QuerySet iterator fill the missing fields.
```

Comment:
@akaariai, could you elaborate on this TODO? Not sure what exactly needs to be done.